sky/server/requests/requests.py

```diff
@@ -703,3 +712,51 @@ def set_request_cancelled(request_id: str) -> None:
     with update_request(request_id) as request_task:
         assert request_task is not None, request_id
         request_task.status = RequestStatus.CANCELLED
+
+
+@init_db
+def clean_finished_requests_with_retention(retention_seconds: int) -> None:
+    """Clean up finished requests older than the retention period.
+
+    This function removes old finished requests (SUCCEEDED, FAILED, CANCELLED)
+    from the database and cleans up their associated log files.
+
+    Args:
+        retention_seconds: Requests older than this many seconds will be
+            deleted.
+    """
+    finished_statuses = [
+        RequestStatus.SUCCEEDED, RequestStatus.FAILED, RequestStatus.CANCELLED
+    ]
+    reqs = get_request_tasks(status=finished_statuses,
+                             created_before=time.time() - retention_seconds)
+
+    for req in reqs:
+        req.log_path.unlink(missing_ok=True)
+
+    id_list_str = ','.join(repr(req.request_id) for req in reqs)
+    assert _DB is not None
+    with _DB.conn:
+        cursor = _DB.conn.cursor()
+        cursor.execute(
+            f'DELETE FROM {REQUEST_TABLE} WHERE request_id IN ({id_list_str})')
```

Comment:
minor: there is an concurrency issue when the `sdk.stream_and_get()` is called during the deletion. The timeline could look like the following
1. The log files are deleted
2. The stream request comes in and try to read the log file as it is still in DB, and fail due to file not found.

We can either allow file not found in stream or set the log file to null before deleting the log_path.